### ISSUE TITLE
allow means to skip roots in`check-types`

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#863]
## Motivation

To allow the likes of this standard JSDoc approach, which allows for documentation per parameter:

```js
/**
 * @param {object} cfg
 * @param {boolean} cfg.isChecked
 * @param {string} cfg.path
 */
```

...but without complaining about "object" not being a specific type if one's `preferredTypes` normally discourage the use of this object. One can already exempt it within a `@typedef`, but one shouldn't wish to normally exempt it in `@param`--only if it is the "root" param. This would therefore need to be a new kind of `exemptedBy`, allowing probably for an object and possibly AST `comment` and `context`.

## Current behavior

Using `object` in a root param when it is prevented in `preferredTypes` will err.

## Desired behavior

Don't err when using `object` in a root param.

## Alternatives considered

N/A
